core/src/main/scala/InsufficientMatingMaterial.scala

```diff
@@ -28,15 +28,74 @@ object InsufficientMatingMaterial:
           }
       )
 
-  /*
+  /**
+   * Returns whether some square in `destinations` can be reached by a king moving from `startSquare`,
+   * while avoiding all squares in `forbidden`.
+   *
+   * `destinations` must not contain `startSquare`, or any square in `forbidden`.
+   */
+  def kingPathExists(startSquare: Square, destinations: Bitboard, forbidden: Bitboard): Boolean =
+    if destinations.intersects(forbidden) || destinations.contains(startSquare) then
+      throw IllegalArgumentException(
+        "`destinations` contains either the start square or some forbidden square"
+      )
```

Comment:
We don't throw exceptions in scala. If the function cannot compute its response type for every combination of its arguments, then it returns a `Try[]`, an `Either[]` or an `Option[]`.